pkg/netbase/ifaceutil/ifaceutil.go

```diff
@@ -0,0 +1,66 @@
+package ifaceutil
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/threefoldtech/zosbase/pkg/network/namespace"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	myceliumPort = "9651"
+)
+
+// GetIPsForIFace retrieves the IP addresses for a given interface name in a specified network namespace.
+// If the namespace name is empty, it retrieves the IP addresses from the host.
+func GetIPsForIFace(iface, nsName string) ([]net.IPNet, error) {
+	getIPs := func(_ ns.NetNS) ([]net.IPNet, error) {
+		var results []net.IPNet
+
+		ln, err := netlink.LinkByName(iface)
+		if err != nil {
+			return nil, err
+		}
+
+		ips, err := netlink.AddrList(ln, netlink.FAMILY_V4)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, ip := range ips {
+			results = append(results, *ip.IPNet)
+		}
+
+		return results, nil
+	}
+
+	if nsName == "" {
+		return getIPs(nil)
+	}
+
+	netns, err := namespace.GetByName(nsName)
+	if err != nil {
+		return nil, err
+	}
+	defer netns.Close()
+
+	var results []net.IPNet
+	err = netns.Do(func(_ ns.NetNS) error {
+		var getErr error
+		results, getErr = getIPs(nil)
```

Comment:
u always pass nil to get ips in both cases if feel here u want to pass netns to get ips and remove netns.Do?